featuretools/tests/feature_discovery/test_feature_discovery.py

```diff
@@ -0,0 +1,361 @@
+                ("f20", Double),
+                ("f21", Boolean),
+                ("f22", BooleanNullable),
+            ],
+            Lag,
+            [["f19", "t_idx"], ["f20", "t_idx"], ["f21", "t_idx"], ["f22", "t_idx"]],
+        ),
+        (
+            [
+                ("idx", Double, {"index"}),
+                ("f23", Double),
+            ],
+            Count,
+            [["idx"]],
+        ),
+        (
+            [
+                ("idx", Double, {"index"}),
+                ("f23", Double),
+            ],
+            AddNumeric,
+            [],
+        ),
+    ],
+)
+@patch.object(LiteFeature, "__lt__", lambda x, y: x.name < y.name)
+def test_get_matching_features(feature_args, primitive, expected):
+    features = [LiteFeature(*args) for args in feature_args]
+    feature_collection = FeatureCollection(features).reindex()
+    actual = get_matching_features(feature_collection, primitive())
+    assert [[y.name for y in x] for x in actual] == expected
+
+
+@pytest.mark.parametrize(
+    "col_defs, primitives, expected",
+    [
+        (
+            [
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Double"),
+            ],
+            [AddNumeric],
+            {"f_1 + f_2", "f_1 + f_4", "f_2 + f_4"},
+        ),
+        (
+            [
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+            ],
+            [Absolute],
+            {"ABSOLUTE(f_1)", "ABSOLUTE(f_2)"},
+        ),
+    ],
+)
+@patch.object(LiteFeature, "__lt__", lambda x, y: x.name < y.name)
+def test_generate_features_from_primitives(col_defs, primitives, expected):
+    input_feature_names = set([x[0] for x in col_defs])
+    df = generate_fake_dataframe(
+        col_defs=col_defs,
+    )
+
+    origin_features = schema_to_features(df.ww.schema)
+    features = generate_features_from_primitives(origin_features, primitives)
+
+    new_feature_names = set([x.name for x in features]) - input_feature_names
+    assert new_feature_names == expected
+
+
+@pytest.mark.parametrize(
+    "col_defs, primitives",
+    [
+        (
+            [
+                ("idx", "Double", {"index"}),
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Boolean"),
+                ("f_5", "Double"),
+            ],
+            [AddNumeric, Absolute, SubtractNumeric, TestDoublePrimitive],
+        ),
+        (
+            [
+                ("idx", "Double", {"index"}),
+                ("t_idx", "Datetime", {"time_index"}),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Boolean"),
+                ("f_5", "Double"),
+            ],
+            [Lag, Day, Absolute, AddNumeric],
+        ),
+    ],
+)
+@patch.object(LiteFeature, "_generate_hash", lambda x: x.name)
+def test_compare_dfs(col_defs, primitives):
```

Comment:
Just wondering if we need a test where we exercise all transform primitives and compare the results so that if we ever make a change to the core DFS algo behavior that might impact some edge cases, we have confirmed we get the same behavior here? Not sure...might be overkill.

Reply:
I like this idea!

Reply:
This is sort of tested in the looking glass tests

Reply:
Ok, I'll put it here, if it becomes a problem we can always remove

Reply:
done!